File: 2020/Day2/Day2.cs
using System;
using System.IO;
using System.Linq;
using System.Collections.Generic;

class MainClass
{
    public static void Main(string[] args)
    {
        List<string> lines = File.ReadLines("./Input.txt").ToList();
        

        Console.WriteLine($"Part1 correct passwords: {Part1(lines)}");
        Console.WriteLine($"Part2 correct passwords: {Part2(lines)}");
    }

    private static int Part1(List<string> lines)
    {
        int totalCorrect = 0;
        foreach(string line in lines)
        {
            string[] args = line.Split(' ');

            int lowerBound = Int32.Parse(args[0].Split('-')[0]);
            int higherBound = Int32.Parse(args[0].Split('-')[1]);

            char checkFor = args[1][0];

            string password = args[2];

            int count = 0;
            foreach (char letter in password) {
                if (letter == checkFor) {
                    count++;
                }
            }

            if (count >= lowerBound && count <= higherBound) {
                totalCorrect++;
            }
        }

        return totalCorrect;
    }

    private static int Part2(List<string> lines)
    {
        int totalCorrect = 0;
        foreach(string line in lines)
        {
            string[] args = line.Split(' ');

            string[] positions = args[0].Split('-');

            char checkFor = args[1][0];

            string password = args[2];

            bool firstCheck = false;
            if (Int32.Parse(positions[0]) - 1 <= password.Length) {
                firstCheck = password[Int32.Parse(positions[0]) - 1] == checkFor ? true : false;
            }
            
            bool secondCheck = false;
            if (Int32.Parse(positions[1]) - 1 <= password.Length) {
                secondCheck = password[Int32.Parse(positions[1]) - 1] == checkFor ? true : false;
            }

            if (firstCheck ^ secondCheck) {
                totalCorrect++;
            }
        }

        return totalCorrect;
    }
}

File: 2020/Day2/Day2.py
with open("Input.txt", "r") as f:
    lines = f.readlines()

def part_1():
    total_correct = 0
    for line in lines:
        args = line.split()

        lower_bound = int(args[0].split("-")[0])
        higher_bound = int(args[0].split("-")[1])

        check_for = args[1][0]

        password = args[2]

        count = 0
        for letter in password:
            if letter == check_for:
                count += 1

        if count >= lower_bound and count <= higher_bound:
            total_correct += 1
    
    return total_correct


def part_2():
    total_correct = 0
    for line in lines:
        args = line.split()

        positions = args[0].split("-")
        check_for = args[1][0]

        password = args[2]

        if int(positions[0]) - 1 <= len(password):
            first_check = True if password[int(positions[0]) - 1] == check_for else False
        else:
            second_check = False

        if int(positions[1]) - 1 <= len(password):
            second_check = True if password[int(positions[1]) - 1] == check_for else False
        else:
            second_check = False

        if first_check ^ second_check:
            total_correct += 1

    return total_correct


print(f"Part1 correct passwords: {part_1()}")
print(f"Part2 correct passwords: {part_2()}")